### DISCUSSION TITLE
`color-modes` should default to true?

### DISCUSSION THREAD

**Discussion by David-Else** [helix-editor/helix#7289] [Ideas] [answered]
```
color-modes | Whether to color the mode indicator with different colors depending on the mode itself | false
```

Why does the setting `color-modes` default to false? Surely if the theme creator added those colors then it should be the default?

Most people won't look for the setting and will never know that the extra feature exists, I only just found it myself in `dark_plus`:

![Screenshot from 2023-06-08 11-39-08](https://github.com/helix-editor/helix/assets/12832280/1cfaaae4-6353-4eec-847d-c78ee8b576ea)

![Screenshot from 2023-06-08 11-39-20](https://github.com/helix-editor/helix/assets/12832280/2f236421-cd30-4aab-aa41-492e91c36e9f)

![Screenshot from 2023-06-08 11-39-34](https://github.com/helix-editor/helix/assets/12832280/ff74ff07-e319-43ec-8d6e-7c1fa166d4ae)

**Comment by the-mikedavis** [comment, marked as answer]
In the long-run I think we should remove the option altogether (so it would behave as always `true`). The `color-modes` option was introduced so that if you found it distracting you could disable it. Now we have theme inheritance though which we can use for the same effect. Similar features like mode-based cursor colors don't have corresponding options and can be disabled by changing the theme.

It would break configs to remove the option though, so I think we should change the default to `true` for now and eventually remove the option when we eventually change the configuration format away from TOML.
